### ISSUE TITLE
Cilium 1.1.2: panic: close of closed channel

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#5158]
Environment: Cilium 1.1.2

```
level=error msg="Error while getting a new service ID. Ignoring service..." error="Error while locking path cilium-net/operational/ServicesV2/SHA256SUMServices/5bcc14a1c1e088876339488e63efded680dc86372aabf83826a03bb672bba414: rpc error: code = Unavailable desc = transport is closing" ipAddr=10.96.0.10 k8sNamespace=kube-system k8sSvcName=kube-dns port=53 protocol=UDP serviceID=dns
level=info msg="Listener disconnected"
level=info msg="Removed listener" count.listener=1
level=info msg="New listener connected." count.listener=2
level=info msg="No request received to manage networking for container" containerID=90dcdd27ed maxRetry=20 subsys=workload-watcher willRetry=false
panic: close of closed channel

goroutine 754 [running]:
github.com/cilium/cilium/pkg/controller.(*Controller).stopController(...)
	/go/src/github.com/cilium/cilium/pkg/controller/controller.go:251
github.com/cilium/cilium/pkg/controller.(*Manager).removeController(0xc42117c9c0, 0xc42369cea0, 0xc4231a3a01)
	/go/src/github.com/cilium/cilium/pkg/controller/manager.go:90 +0x43
github.com/cilium/cilium/pkg/controller.(*Manager).UpdateController(0xc42117c9c0, 0xc4231a3ae0, 0x4c, 0xc422368c80, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/cilium/cilium/pkg/controller/manager.go:65 +0x404
main.(*Daemon).updateCiliumNetworkPolicyV2(0xc42036e7e0, 0x2909ec0, 0xc42357b7c0, 0xc420291400, 0xc4202917c0)
	/go/src/github.com/cilium/cilium/daemon/k8s_watcher.go:1554 +0x30b
main.(*Daemon).EnableK8sWatcher.func14.1(0x0, 0xc4235c0120)
	/go/src/github.com/cilium/cilium/daemon/k8s_watcher.go:425 +0x4e
github.com/cilium/cilium/pkg/serializer.(*functionQueue).run(0xc42358a8c0)
	/go/src/github.com/cilium/cilium/pkg/serializer/func_queue.go:65 +0x70
created by github.com/cilium/cilium/pkg/serializer.NewFunctionQueue
	/go/src/github.com/cilium/cilium/pkg/serializer/func_queue.go:45 +0xbe
level=fatal msg="Agent pipe unexpectedly closed, shutting down"
```

**Comment by joestringer** [comment]
Fixing this may be a matter of backporting #4683. @rlenglet and I discussed briefly on slack.

**Comment by tgraf** [comment]
Issue resolved via #4683
